documentation/src/test/java/org/hibernate/userguide/mapping/basic/BigIntegerMappingTests.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.userguide.mapping.basic;
+
+import java.math.BigInteger;
+import java.sql.Types;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.hibernate.metamodel.MappingMetamodel;
+import org.hibernate.metamodel.mapping.JdbcMapping;
+import org.hibernate.metamodel.mapping.internal.BasicAttributeMapping;
+import org.hibernate.persister.entity.EntityPersister;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Tests for mapping `BigInteger` values
+ *
+ * @author Steve Ebersole
+ */
+@DomainModel( annotatedClasses = BigIntegerMappingTests.EntityOfBigIntegers.class )
+@SessionFactory
+public class BigIntegerMappingTests {
+
+	@Test
+	public void testMappings(SessionFactoryScope scope) {
+		// first, verify the type selections...
+		final MappingMetamodel domainModel = scope.getSessionFactory().getDomainModel();
+		final EntityPersister entityDescriptor = domainModel.findEntityDescriptor( EntityOfBigIntegers.class );
+
+		{
+			final BasicAttributeMapping attribute = (BasicAttributeMapping) entityDescriptor.findAttributeMapping( "wrapper" );
+			assertThat( attribute.getJavaTypeDescriptor().getJavaTypeClass(), equalTo( BigInteger.class ) );
+
+			final JdbcMapping jdbcMapping = attribute.getJdbcMapping();
+			assertThat( jdbcMapping.getJavaTypeDescriptor().getJavaTypeClass(), equalTo( BigInteger.class ) );
+			assertThat( jdbcMapping.getJdbcTypeDescriptor().getJdbcTypeCode(), is( Types.NUMERIC ) );
+		}
+
+
+		// and try to use the mapping
+		scope.inTransaction(
+				(session) -> session.persist( new EntityOfBigIntegers( 1, BigInteger.TWO ) )
```

Comment:
`BigInteger.TWO` causes a compilation error with java 8, it has been introduced with java 9

Reply:
Does that really cause a problem?  I thought we allowed Java 9 syntax in tests?